### DISCUSSION TITLE
When creating a record, can I use objects directly as in Where?

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [prisma/prisma#14752] [Q&A] [answered]
In case of a Where query I can directly specify objects for relation fields, here for example Owner
"user" in this case is an actual user returned from  the prisma client.
```javascript
await client.artifact.count({where: {Owner: user, Name: originalName}})
```
However, this is not possible in the Update or Create case, specifically here the following fields are affected:  Epoch, Owner, and Category.
I´m getting my data like this, here is an example for the "requiredEpoch":
```javascript
const requiredEpoch = await client.epoch.findUnique({where: {ID: parseInt(epoch)}}) ?? null;
```
And then i try to use it

```javascript
 await client.excavationSite.create({data:{
                    Name:name,
                        Epoch:requiredEpoch,
                        Latitude:parseFloat(latitude),
                        Longitude:parseFloat(longitude),
                        Note:notes,
                        Owner:user,
                        Category:await client.excavationSiteCategory.findUniqueOrThrow({where:{ID:parseInt(category)}})
                    }})
```
![image](https://user-images.githubusercontent.com/46539758/184021044-13ffea50-b854-4767-b9ad-e6c9de73aa43.png)

The error i´m getting says the same thing about every field which has a red underline:
```
TS2559: Type 'Epoch' has no properties in common with type 'EpochCreateNestedOneWithoutExcavationSiteInput'.
```

I have no problem doing this via the ID's, however I would just be interested to know if this is intended from Prisma that you have to access the ID here instead of being able to give real objects, or if I am doing something wrong here.

Thanks in advance!

**Comment by nurul3101** [comment, marked as answer]
Hi @manuelpoelzl 👋 
So if I am understanding correctly, your question is if you can pass an epoch response that you obtained from the `findUnique` query in a create query, right?

**Reply by manuelpoelzl** [reply]
Hey!
yes exactly.
I mean, I think access via ID's is perfectly ok, so that would just be an added benefit of being able to pass the object directly.

**Reply by pantharshit00** [reply]
I think you need to pass the data under `create` key inside of `Epoch` object:

```ts
await client.excavationSite.create({data:{
  Name: name,
  Epoch:{
    create: requiredEpoch
   }
  // ..
  // ..
  }
})
```

If this doesn't work, can you please share your datamodel here?

**Reply by pantharshit00** [reply]
Hello @manuelpoelzl 

Just wanted to make sure if you were able to solve this. Can you please confirm if this issue is still relevant to you?

**Reply by manuelpoelzl** [reply]
Hi!
Yes i was able to solve this, so this can be closed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
